arangod/Aql/Functions.cpp

```diff
@@ -9211,6 +9215,41 @@ AqlValue Functions::MakeDistributeGraphInput(
   return AqlValue{input};
 }
 
+#ifdef USE_ENTERPRISE
+AqlValue Functions::SelectSmartDistributeGraphInput(
+    arangodb::aql::ExpressionContext* expressionContext, AstNode const&,
+    VPackFunctionParameters const& parameters) {
+  AqlValue const& from = extractFunctionParameterValue(parameters, 0);
+  VPackSlice input = from.slice();  // will throw when wrong type
+  if (ADB_UNLIKELY(!input.isObject() ||
+                   !input.hasKey(StaticStrings::IdString) ||
```

Comment:
This line is not necessary. The following `input.get(...).isString()` will always be false if the attribute is not present. In fact, `slice.get(attribute)` will return a Slice of type None if the attribute does not exist. 
```suggestion
```

Reply:
good point